### DISCUSSION TITLE
Sanitize Form Input Before Validate

### DISCUSSION THREAD

**Discussion by jackjackde** [filamentphp/filament#13592] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.63

### How can we help you?

Hi,

i'm currently struggling with sanitizing the form input values. I'm looking for a solution which use a hook or event, which is fired before validation. The current situation looks like this:

I stored an array in my model, which contains the attribute names as a key and the sanitize rules as the value:
```php
// Model
protected array $sanitize = [
    'name' => 'trim|capitalize|...',
    ....
];
```
I've tried to override the `prepareForValidation` method but with the `HasState::dehydrateState` method, the state retrieve the old data :\
i want to sanitize the given form input data / state before validating. The sanitize function returns the sanitized data as an array so i'm looking for a hook or event on form level, not field level. 
Can someone give me a hint, please, how to solve it? thanks a lot!

**Comment by leandrocfe** [comment]
You can use `beforeValidate()` https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks

```php
protected function beforeValidate(): void
{
        $this->data['name'] = ...
}
```

**Reply by jackjackde** [reply]
thanks, but i'm not using the panels, it's just the "form in livewire component" way. sorry, that's something i forgot to say

**Reply by jackjackde** [reply]
i missed that point, that i can work with the $data variable in the live component, my fault. So i got a solution but it's not so nice as i hoped. 
thanks anyway!

**Reply by leandrocfe** [reply, marked as answer]
You can sanitize this before `$this->form->getState()`
